### PR TITLE
no more offset  in src/rec, model takes care of it + compiler version…

### PR DESCRIPTION
**Pull request by mloubout**
Removed all "offset" in inject/interpolate, let model take care of origin with pml.

Tiny fix for compiler version because for example intel names its version in a way `version` does not recognize as valid.

### CODE REVIEW

**Comment by FabioLuporini** [comment]
fantastic

File: devito/compiler.py [hunk 1]
```diff
@@ -49,11 +49,13 @@ def sniff_compiler_version(cc):
         try:
             # gcc-7 series only spits out patch level on dumpfullversion.
             ver = check_output([cc, "-dumpfullversion"], stderr=DEVNULL).decode("utf-8")
-            ver = version.StrictVersion(ver.strip())
+            ver = '.'.join(ver.strip().split('.')[:3])
```
[the remark this answers is not in the document]
Comment (mloubout) [reply devito/compiler.py]: Yes it is , it only turns e.g `1.2.3.12345` into `1.2.3` as `version` only accept three values.

File: examples/seismic/model.py [hunk 2]
```diff
@@ -459,13 +459,14 @@ def __init__(self, origin, spacing, shape, space_order, nbpml=20,
                  dtype=np.float32):
         self.shape = shape
         self.nbpml = int(nbpml)
-        self.origin = tuple([dtype(o) for o in origin])
+        self.origin = tuple([dtype(o ) for o in origin])
```
[the remark this answers is not in the document]
Comment (mloubout) [reply examples/seismic/model.py]: Left it as it was, yes flake 8 will complai n will just amend push.

File: examples/seismic/model.py [hunk 3]
```diff
@@ -459,13 +459,14 @@ def __init__(self, origin, spacing, shape, space_order, nbpml=20,
...
 
+        origin_pml = tuple([dtype(o - s*nbpml) for o, s in zip(origin, spacing)])
```
[the remark this answers is not in the document]
Comment (mloubout) [reply examples/seismic/model.py]: done

**Pull request merged**